rust_src/src/lisp.rs

```diff
@@ -882,6 +882,12 @@ impl LispObject {
         )
     }
 
+    pub fn as_marker_or_error(self) -> LispMarkerRef {
+        self.as_marker().unwrap_or_else(
+            || wrong_type!(Qmarkerp, self),
```

Comment:
closure is not necessary. you can use `self.as_marker().unwrap_or(wrong_type!(Qmarkerp, self))`